### ISSUE TITLE
`MvNormal` and `Tracker`

### ISSUE THREAD

**Issue by willtebbutt** [TuringLang/Turing.jl#717]
We still can't compute the `logpdf` of a multivariate Normal distribution and backprop using `Tracker`, which is a bit of a shame. Here's a MWE:
```julia
]dev Tracker
using Tracker, PDMats, LinearAlgebra

A = randn(3, 3);
B = A * A' + I;
x, m = randn(3), randn(3), B;

y, back = Tracker.forward((x, m, B)->logpdf(MvNormal(m, PDMat(B)), x), x, m, B)
ERROR: MethodError: no method matching PDMat(::TrackedArray{…,Array{Float64,2}})
Closest candidates are:
  PDMat(::AbstractArray{T,2} where T, ::Cholesky) at /home/wct23/.julia/dev/PDMats/src/pdmat.jl:13
  PDMat(::Array{T,2} where T) at /home/wct23/.julia/dev/PDMats/src/pdmat.jl:19
  PDMat(::Symmetric) at /home/wct23/.julia/dev/PDMats/src/pdmat.jl:20
  ...
Stacktrace:
 [1] (::getfield(Main, Symbol("##7#8")))(::TrackedArray{…,Array{Float64,1}}, ::TrackedArray{…,Array{Float64,1}}, ::TrackedArray{…,Array{Float64,2}}) at ./REPL[13]:1
 [2] #20 at /home/wct23/.julia/dev/Tracker/src/back.jl:148 [inlined]
 [3] forward(::getfield(Tracker, Symbol("##20#22")){getfield(Main, Symbol("##7#8"))}, ::Params) at /home/wct23/.julia/dev/Tracker/src/back.jl:135
 [4] forward(::Function, ::Array{Float64,1}, ::Array{Float64,1}, ::Vararg{Any,N} where N) at /home/wct23/.julia/dev/Tracker/src/back.jl:148
 [5] top-level scope at none:0
```
I'm not entirely sure what the issue _actually_ is here if I'm honest.

(Note that `Tracker` has recently been split off from `Flux` in preparation for `Zygote`, presumably becoming the default AD, hence the lack of `using Flux.Tracker`.)

**Comment by trappmartin** [comment]
This is a recurring issue I observed when using Tracker. A dirty fix for this is to manually ensure all `TrackedArray`s are replaced by `Array`s of `TrackedReal`s.  I suppose this, however, has bad performance. (I haven't checked.)

MWE:
```julia
using Tracker, PDMats, LinearAlgebra, Distributions

A = randn(3, 3);
B = A * A' + I;
x, m = randn(3), randn(3), B;

y, back = Tracker.forward((x, m, B)->logpdf(MvNormal([i for i in m], PDMat([i for i in B])), [i for i in x]), x, m, B)
back(1)[1]
Tracked 3-element Array{Float64,1}:
 0.3447760194226308
 0.07158986201398362
 1.2737932266770664
```

**Comment by tszanalytics** [comment]
I'm still unable to use the MVNormal in an @model macro.  This is apparently something that changed with the recent julia update.  There is a recent fix in Mamba.jl to address this:
"Distributions compatibility fixes for Categorical, MvNormal, and MvNormalCanon constructors."

Any chance a similar fix could be implemented for Turing.jl?  It seemed to be a pretty quick fix for Mamba - I emailed Brian Smith asking about it and he implemented a fix within a few days, so hopefully it should be fairly straightforward.

**Comment by mohamed82008** [comment]
Fixed with #757.
